google/cloud/forseti/services/explain/service.py

```diff
@@ -79,10 +97,16 @@ def ListResources(self, request, context):
         Returns:
             object: proto message of list of resources
         """
+        reply = explain_pb2.ListResourcesReply()
+
+        if not self.is_supported:
+            context.set_code(StatusCode.FAILED_PRECONDITION)
```

Comment:
You could create a small private helper method and use that.

```
def _set_not_supported_status(context, reply):
  context.set_code(..)
  context.set_details(..)
  return reply
```

Then in you can just do:

```
if not self.is_supported:
  return _set_not_supported_status(context, reply)
```